gatorgrade/input/command_line_generator.py

```diff
@@ -0,0 +1,66 @@
+"""Generates a dictionary of shell and gator grader command options from a list of dict checks."""
+
+import os
+
+
+# Function to generate command options from a list of object checks
+
+# pylint: disable=too-many-nested-blocks
+def generate_checks(file_context_checks):
+    """Generate a dictionary of checks based on the configuration file.
+
+        This dictionary will have the format:
+        {
+            "shell": List of shell checks,
+            "gatorgrader": List of GatorGrader checks
+        },
+    Args:
+        file_context_checks: List containing dictionaries that contain file contexts
+            (either a file path or None if no file context) and checks in another dictionary
+            (can be either GatorGrader or shell checks).
+            The input list is generated based on the configuration file.
+    """
+    gatorgrader_checks = []
+    shell_checks = []
+    for file_context_check in file_context_checks:
+        # assigning the check from the dict object
+        check = file_context_check["check"]
+        # If the check has a 'command', then it is a shell check
+        if "command" in check:
+            shell_checks.append(check)
+        # Else it's a gator grader check
+        else:
+            gatorgrader_command_options = []
+            # Defining the description and option
+            description = check["description"]
+            options = check["options"]
+            # Creating a list that has description, check, and options in it for GatorGrader check
+            gatorgrader_command_options = ["--description", f"{description}"]
```

Comment:
It's possible for a check to have no `description` -- the only required attribute is `check`. So, you should put this line in an `if` statement just as you have with `options`.

Reply:
Good point, I just added an if statement if description key exists.